data/mods/ssb/moves.ts

```diff
@@ -2815,1775 +3171,2563 @@ export const Moves: {[k: string]: ModdedMoveData} = {
-	},
-	// Teclis
-	absoluteconfiguration: {
-		accuracy: true,
-		basePower: 0,
-		category: "Status",
-		desc: "Puts the foe to sleep. Summons a Nightmare Field with the effects of Nightmare for 4 turns.",
-		shortDesc: "Puts foe to sleep. Nightmare for 4 turns.",
-		name: "Absolute Configuration",
-		isNonstandard: "Custom",
-		pp: 1,
-		priority: 0,
-		flags: {reflectable: 1},
-		onTryMove() {
-			this.attrLastMove('[still]');
-		},
-		onPrepareHit(target, source) {
-			this.add('-anim', source, 'Dark Pulse', target);
-			this.add('-anim', source, 'Dark Void', target);
-		},
-		onHit(source) {
-			this.field.addPseudoWeather('nightmarefield', source);
-		},
-		status: 'slp',
-		isZ: "darkrainiumz",
-		target: "normal",
-		type: "Dark",
+		type: "Ghost",
 	},
-	// Used for Teclis's z-move
-	nightmarefield: {
+
+	// for RavioliQueen's ability
+	pitchblackterrain: {
 		accuracy: true,
 		basePower: 0,
 		category: "Status",
-		desc: "For 4 turns, Nightmare Field is active. During the effect, sleeping Pokemon suffer from the effects of Nightmare.",
-		shortDesc: "4 turns. Sleeping Pokemon suffer Nightmare.",
-		name: "Nightmare Field",
+		desc: "For 5 turns, Non Ghost types take 1/16th damage; Has boosting effects on Mismagius.",
+		shortDesc: "5 turns. Non Ghost types take 1/16th damage; Has boosting effects on Mismagius.",
+		name: "Pitch Black Terrain",
 		isNonstandard: "Custom",
+		gen: 8,
 		pp: 10,
 		priority: 0,
 		flags: {},
-		pseudoWeather: 'nightmarefield',
+		terrain: 'pitchblackterrain',
 		condition: {
-			duration: 4,
+			duration: 5,
+			durationCallback(source, effect) {
+				if (source?.hasItem('terrainextender')) {
+					return 8;
+				}
+				return 5;
+			},
+			onHit(target, source, move) {
+				if (!target.hp || target.species.name !== 'Mismagius') return;
+				if (move?.effectType === 'Move' && move.category !== 'Status') {
+					if (this.boost({spe: 1}, target)) {
+						this.add('-message', `${target.name} got a boost by the terrain!`);
+					}
+				}
+			},
+			onSwitchInPriority: -1,
+			onSwitchIn(target) {
+				if (target?.species.name === 'Mismagius') {
+					if (this.boost({spa: 1, spd: 1}, target)) {
+						this.add('-message', `${target.name} got a boost by the terrain!`);
+					}
+				}
+			},
 			onStart(battle, source, effect) {
-				if (effect && effect.effectType === 'Ability') {
-					this.add('-fieldstart', 'move: Nightmare Field', '[from] ability: ' + effect, '[of] ' + source);
+				if (effect?.effectType === 'Ability') {
+					this.add('-fieldstart', 'move: Pitch Black Terrain', '[from] ability: ' + effect, '[of] ' + source);
 				} else {
-					this.add('-fieldstart', 'move: Nightmare Field');
+					this.add('-fieldstart', 'move: Pitch Black Terrain');
+				}
+				this.add('-message', 'The battlefield became dark!');
+				if (source?.species.name === 'Mismagius') {
+					if (this.boost({spa: 1, spd: 1}, source)) {
+						this.add('-message', `${source.name} got a boost by the terrain!`);
+					}
 				}
```

Comment:
```suggestion
				if (source?.species.name === 'Mismagius' && this.boost({spa: 1, spd: 1}, source)) {
					this.add('-message', `${source.name} got a boost by the terrain!`);
				}
```

Reply:
I'd rather not condense here, ill just return prematurely if conditions arent met instead of using nested ifs

Reply:
it just feels more readable imo

Reply:
Sounds good!